bilby/gw/conversion.py

```diff
@@ -259,7 +259,7 @@ def convert_to_lal_binary_black_hole_parameters(parameters):
                         )
                         converted_parameters[f"cos_tilt_{idx}"] = 1.0
 
-    for key in ["phi_jl", "phi_12"]:
+    for key in ["phi_jl", "phi_12", "eccentricity", "mean_per_ano"]:
```

Comment:
Thanks for this PR! My personal preference would be to not have the `eccentricity` and `mean_per_ano` fields in quasicircular PE files, so I would vote for not including these here at all. But I'd also like to hear opinions from others.

Reply:
Naively, I would have thought that eccentric PE with `gwsignal_binary_black_hole` would work even without changes to this line. Am I missing something?

Reply:
I introduced eccentricity and mean anomaly here because with the addition of eccentricity and mean_per_ano in the arguments of the `gwsignal_binary_black_hole` function in source.py, when one tries to run a quasicircular approximant (which does not specify eccentricity and mean_per_ano) Bilby would crash.  Adding them here is a possible fix, but if you have other suggestions we are happy to modify the code

Reply:
Thanks for explaining. I still think there are other, better ways of dealing with this. Here are some in decreasing order of preference:

1. Refactor the current code by adding a` _base_gwsignal_cbc_fd_waveform` function similar to `_base_lal_cbc_fd_waveform`, and make `gwsignal_binary_black_hole` call this function similar to how `lal_binary_black_hole` called `_base_lal_cbc_fd_waveform`. Write another function `gwsignal_eccentric_binary_black_hole` which also calls `_base_gwsignal_cbc_fd_waveform`. This change will also make things easier if e.g. you want to add e.g. tides to one of the waveforms.

2. If for some reason 1 isn't preferred, I think the easiest way to not make bilby crash is to fix `eccentricity` and `mean_per_ano` to zero in your prior.

3. You could also set a default value of `eccentricity` and `mean_per_ano` in the function definition.

I think 1 is by far the most robust option. Tagging @asb5468 @ColmTalbot @mj-will for opinions.

Reply:
Thank you @adivijaykumar for the suggestions: is [this](https://github.com/AntoniRamosBuades/bilby/commit/1b3527cb85892002081194dc7db31fda64d17a81) addressing it? If yes, I'll continue from there and revert the other changes and tests.

Concerning 2/ and 3/, I would like to avoid default parameters as they may lead to silent mistakes.

Reply:
@raffienficiaud Yes I think that addresses it for the most part! You might want to take out the default waveform_kwargs from inside the `_base_*` function and put it into the individual functions.

Reply:
I reverted the changes in this file and the related unit test.